### DISCUSSION TITLE
SetPositions for Conformer

### DISCUSSION THREAD

**Discussion by pultar** [rdkit/rdkit#6147] [Ideas]
It is currently possible to get individual or all atomic positions for a conformer (`GetAtomPosition` and `GetPositions`) as `Point3D` or `np.array`  but I could only find the equivalent function `SetAtomPosition` for setting individual atomic positions.

Would it be desirable to have a function `SetPositions` that takes in a `np.array` and updates coordinates in one go? This could be useful for e.g. updating positions of a conformer from an xyz file after a QM geometry optimization. It could also potentially make manual set ups of conformers like here easier:

https://github.com/rdkit/rdkit/blob/b53ee20b49bb3dce24486b73fa1ef7d69f9a9e84/rdkit/Chem/Features/UnitTestFeatDirUtilsRD.py#L13

I guess the function would be a combination of these two functions here?

https://github.com/rdkit/rdkit/blob/1f4584b2ca07037701dd23fea854943bb1381892/Code/GraphMol/Wrap/Conformer.cpp#L56

and 

https://github.com/rdkit/rdkit/blob/1f4584b2ca07037701dd23fea854943bb1381892/Code/GraphMol/Wrap/Conformer.cpp#L33

with wrapping like here:

https://github.com/rdkit/rdkit/blob/1f4584b2ca07037701dd23fea854943bb1381892/Code/GraphMol/Wrap/Conformer.cpp#L88

Let me know what you think.

**Comment by greglandrum** [comment]
Hi @pultar, I agree that this definitely could be a useful piece of functionality to have available.
It's worth considering whether the new `SetPositions()` function needs to take an `np.array` or if it would just make sense to take a sequence of sequences of floats so that you could a list of lists, or tuple of tuples (or list of tuples, or...). If this complicates the code too much then it's probably not worth doing since doing the conversion in Python is a one liner, but if it's simple...
